MP-4508: UI-Components doc (#321)
spryker/spryker-docs: docs/marketplace/dev/front-end/ui-components-library/index.md

---
title: UI Components Library
description: This articles provides details about ui components library.
template: concept-topic-template
---

This articles provides details about ui components library.

## Introduction

There is a set of UI Angular Components that is built separately from the Spryker Core and distributed via NPM Registry independently.
Each package can be installed via npm command `npm install @spryker/package_name` or yarn `yarn add @spryker/package_name`.

> UI Angular Components lib has `Angular 9`, `rxjs`, `ng-zorro-antd 9` under the hood.

Reviewer: ```suggestion
UI Angular Components is built using Angular 9, rxjs and Ant Design under the hood.
```

  - @spryker/data-transformer.collate ([npm](https://www.npmjs.com/package/@spryker/data-transformer.collate))
  - @spryker/data-transformer.configurator.table ([npm](https://www.npmjs.com/package/@spryker/data-transformer.configurator.table))
  - @spryker/data-transformer.date-parse ([npm](https://www.npmjs.com/package/@spryker/data-transformer.date-parse))
  - @spryker/data-transformer.date-serialize ([npm](https://www.npmjs.com/package/@spryker/data-transformer.date-serialize))
  - @spryker/data-transformer.filter.equals ([npm](https://www.npmjs.com/package/@spryker/data-transformer.filter.equals))
  - @spryker/data-transformer.filter.range ([npm](https://www.npmjs.com/package/@spryker/data-transformer.filter.range))
  - @spryker/data-transformer.filter.text ([npm](https://www.npmjs.com/package/@spryker/data-transformer.filter.text))
  - @spryker/data-transformer.lens ([npm](https://www.npmjs.com/package/@spryker/data-transformer.lens))
  - @spryker/data-transformer.object-map ([npm](https://www.npmjs.com/package/@spryker/data-transformer.object-map))
  - @spryker/data-transformer.pluck ([npm](https://www.npmjs.com/package/@spryker/data-transformer.pluck))

> - @spryker/datasource ([npm](https://www.npmjs.com/package/@spryker/datasource))
> - @spryker/datasource.inline ([npm](https://www.npmjs.com/package/@spryker/datasource.inline), [story](https://spy-storybook.web.app/?path=/story/datasourceinline--primary))

Reviewer 2: ```suggestion
- @spryker/datasource ([npm](https://www.npmjs.com/package/@spryker/datasource))
- @spryker/datasource.http ([npm](https://www.npmjs.com/package/@spryker/datasource.http))
- @spryker/datasource.inline ([npm](https://www.npmjs.com/package/@spryker/datasource.inline), [story](https://spy-storybook.web.app/?path=/story/datasourceinline--primary))
```

- @spryker/datasource.inline.table ([npm](https://www.npmjs.com/package/@spryker/datasource.inline.table), [story](https://spy-storybook.web.app/?path=/story/tabledatasourceinlineservice--with-table))
- @spryker/date-picker ([npm](https://www.npmjs.com/package/@spryker/date-picker), [story](https://spy-storybook.web.app/?path=/story/datepickercomponent--primary))
- @spryker/drawer ([npm](https://www.npmjs.com/package/@spryker/drawer), [story](https://spy-storybook.web.app/?path=/story/drawerscomponent--primary))
- @spryker/dropdown ([npm](https://www.npmjs.com/package/@spryker/dropdown), [story](https://spy-storybook.web.app/?path=/story/dropdowncomponent--primary))
- @spryker/form-item ([npm](https://www.npmjs.com/package/@spryker/form-item), [story](https://spy-storybook.web.app/?path=/story/formitemcomponent--primary))
- @spryker/grid ([npm](https://www.npmjs.com/package/@spryker/grid), [story](https://spy-storybook.web.app/?path=/story/gridcomponent--primary))
- @spryker/header ([npm](https://www.npmjs.com/package/@spryker/header), [story](https://spy-storybook.web.app/?path=/story/headercomponent--primary))

> - @spryker/headline ([npm](https://www.npmjs.com/package/@spryker/headline), [story](https://spy-storybook.web.app/?path=/story/headlinecomponent--primary))
> - @spryker/headline ([npm](https://www.npmjs.com/package/@spryker/headline), [story](https://spy-storybook.web.app/?path=/story/htmlrenderercomponent--with-static-html))

Reviewer 2: ```suggestion
- @spryker/headline ([npm](https://www.npmjs.com/package/@spryker/headline), [story](https://spy-storybook.web.app/?path=/story/headlinecomponent--primary))
- @spryker/html-renderer ([npm](https://www.npmjs.com/package/@spryker/html-renderer), [story](https://spy-storybook.web.app/?path=/story/htmlrenderercomponent--with-static-html))
```
